handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslClientContext.java

```diff
@@ -0,0 +1,288 @@
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedKeyManager;
+import javax.net.ssl.X509ExtendedTrustManager;
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+public final class ReleasableOpenSslClientContext extends ReleasableOpenSslContext {
+    private static final InternalLogger logger =
+            InternalLoggerFactory.getInstance(ReleasableOpenSslClientContext.class);
+    private final OpenSslSessionContext sessionContext;
+
+    ReleasableOpenSslClientContext(X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
+                         X509Certificate[] keyCertChain, PrivateKey key, String keyPassword,
+                         KeyManagerFactory keyManagerFactory, Iterable<String> ciphers,
+                         CipherSuiteFilter cipherFilter, ApplicationProtocolConfig apn,
+                         long sessionCacheSize, long sessionTimeout)
+            throws SSLException {
+        super(ciphers, cipherFilter, apn, sessionCacheSize, sessionTimeout, SSL.SSL_MODE_CLIENT, keyCertChain,
+              ClientAuth.NONE, true);
+        boolean success = false;
+        try {
+            sessionContext = newSessionContext(this, ctx, engineMap, trustCertCollection, trustManagerFactory,
+                                               keyCertChain, key, keyPassword, keyManagerFactory);
+            success = true;
+        } finally {
+            if (!success) {
+                release();
+            }
+        }
+    }
+
+    @Override
+    OpenSslKeyMaterialManager keyMaterialManager() {
+        return null;
+    }
+
+    @Override
+    public OpenSslSessionContext sessionContext() {
+        return sessionContext;
+    }
+
+    static OpenSslSessionContext newSessionContext(ReleasableOpenSslContext thiz, long ctx, OpenSslEngineMap engineMap,
+                                                   X509Certificate[] trustCertCollection,
+                                                   TrustManagerFactory trustManagerFactory,
+                                                   X509Certificate[] keyCertChain, PrivateKey key, String keyPassword,
+                                                   KeyManagerFactory keyManagerFactory) throws SSLException {
+        if (key == null && keyCertChain != null || key != null && keyCertChain == null) {
+            throw new IllegalArgumentException(
+                    "Either both keyCertChain and key needs to be null or none of them");
+        }
+        synchronized (ReleasableOpenSslContext.class) {
+            try {
+                if (!OpenSsl.supportsKeyManagerFactory()) {
+                    if (keyManagerFactory != null) {
+                        throw new IllegalArgumentException(
+                                "KeyManagerFactory not supported");
+                    }
+                    if (keyCertChain != null && key != null) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)